File: biteRush-backend/menu-service/src/main/java/com/biterush/menu_service/service/MenuService.java
package com.biterush.menu_service.service;

import com.biterush.menu_service.model.dto.request.MenuCategoryRequestDTO;
import com.biterush.menu_service.model.dto.request.MenuItemRequestDTO;
import com.biterush.menu_service.model.entity.MenuCategory;
import com.biterush.menu_service.model.entity.MenuItem;

import java.util.List;
import java.util.UUID;

public interface MenuService {
    String createMenuCategory(MenuCategoryRequestDTO requestDTO);
    String createMenuItem(MenuItemRequestDTO requestDTO);
    String updateMenuItem(UUID itemId,MenuItemRequestDTO requestDTO);
    String updateMenuCategory(UUID categoryId,MenuCategoryRequestDTO requestDTO);
    String deleteMenuItem(UUID itemId);
    String deleteMenuCategory(UUID categoryId);
    MenuItem getMenuItemById(UUID itemId);
    List<MenuItem> getMenuItems(UUID restaurantId);
//    List<MenuItem> getMenuItemsByCategory(UUID categoryId);
    List<MenuCategory> getMenuCategories(UUID restaurantId);
    MenuCategory getMenuCategory(UUID categoryId);
}

File: biteRush-frontend/src/services/MenuService.js
import { apiInstance } from "./AxiosInstance"

export const menuService = {

    getMenu(restaurantId){
        return apiInstance.get(`/restaurants/${restaurantId}/menu`)
    },

    createMenuCategory(restaurantId,menuCategoryJSON){
        return apiInstance.post(`/restaurants/${restaurantId}/menu/categories`,menuCategoryJSON)
    },

    updateMenuCategory(categoryId,menuCategoryJSON){
        return apiInstance.put(`menu/categories/${categoryId}`,menuCategoryJSON)
    },

    deleteMenuCategory(categoryId){
        return apiInstance.delete(`/menu/categories/${categoryId}`)
    },

    createMenuItem(restaurantId,menuItemJSON){
        return apiInstance.post(`/restaurants/${restaurantId}/menu/items`,menuItemJSON)
    },

    getMenuItem(itemId){
        return apiInstance.get(`/menu/items/${itemId}`)
    },

    updateMenuItem(itemId,menuItemJSON){
        return apiInstance.put(`/menu/items/${itemId}`,menuItemJSON)
    }
}